static/js/message_events.js

```diff
@@ -64,14 +64,18 @@ exports.insert_new_messages = function insert_new_messages(messages, sent_by_thi
     message_util.add_new_messages(messages, message_list.all);
 
     let render_info;
+    let opts;
 
     if (narrow_state.active()) {
         // We do this NOW even though the home view is not active,
         // because we want the home view to load fast later.
         message_util.add_new_messages(messages, home_msg_list);
 
         if (narrow_state.filter().can_apply_locally()) {
-            render_info = message_util.add_new_messages(messages, message_list.narrowed);
+            if (!current_msg_list.fetch_status.has_found_newest()) {
```

Comment:
Let's add a comment in here: "We can't render new messages if we haven't found the latest messages in the view without making it appear the message was just after previously rendered messages when that's inaccurate."